azure/services/inboundnatrules/client.go

```diff
@@ -18,175 +18,134 @@ package inboundnatrules
 
 import (
 	"context"
-	"encoding/json"
-	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
-	"github.com/Azure/go-autorest/autorest"
-	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
 	"github.com/pkg/errors"
-	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/asyncpoller"
 	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // client wraps go-sdk.
 type client interface {
-	List(context.Context, string, string) (result []network.InboundNatRule, err error)
+	List(context.Context, string, string) (result []armnetwork.InboundNatRule, err error)
 	Get(context.Context, azure.ResourceSpecGetter) (result interface{}, err error)
-	CreateOrUpdateAsync(context.Context, azure.ResourceSpecGetter, interface{}) (result interface{}, future azureautorest.FutureAPI, err error)
-	DeleteAsync(context.Context, azure.ResourceSpecGetter) (future azureautorest.FutureAPI, err error)
-	IsDone(context.Context, azureautorest.FutureAPI) (isDone bool, err error)
-	Result(context.Context, azureautorest.FutureAPI, string) (result interface{}, err error)
+	CreateOrUpdateAsync(context.Context, azure.ResourceSpecGetter, string, interface{}) (result interface{}, poller *runtime.Poller[armnetwork.InboundNatRulesClientCreateOrUpdateResponse], err error)
+	DeleteAsync(context.Context, azure.ResourceSpecGetter, string) (poller *runtime.Poller[armnetwork.InboundNatRulesClientDeleteResponse], err error)
```

Comment:
Not strictly related to these changes, but it looks like `List` is the only method called on the interface type, so I think the others can be removed.